src/Emulator/Peripherals/Peripherals/DMA/STM32DMA2D.cs

```diff
@@ -56,18 +59,31 @@ public long Size
         private STM32DMA2D()
         {
             var controlRegister = new DoubleWordRegister(this);
-            startFlag = controlRegister.DefineFlagField(0, FieldMode.Read | FieldMode.Write, name: "Start", changeCallback: (old, @new) => { if(@new) DoTransfer(); });
+            startFlag = controlRegister.DefineFlagField(0, FieldMode.Read | FieldMode.Write, name: "Start", writeCallback: (old, @new) => { 
+                if(@new) DoTransfer(); 
+            });
             dma2dMode = controlRegister.DefineEnumField<Mode>(16, 2, FieldMode.Read | FieldMode.Write, name: "Mode");
 
             var foregroundClutMemoryAddressRegister = new DoubleWordRegister(this).WithValueField(0, 32, FieldMode.Read | FieldMode.Write);
             var backgroundClutMemoryAddressRegister = new DoubleWordRegister(this).WithValueField(0, 32, FieldMode.Read | FieldMode.Write);
 
-            var interruptFlagClearRegister = new DoubleWordRegister(this).WithFlag(1, FieldMode.Read | FieldMode.WriteOneToClear, name: "CTCIF", changeCallback: (old, @new) => { if(!@new) IRQ.Unset(); });
+            var interruptStatusRegister = new DoubleWordRegister(this);
+            interruptStatusRegister.DefineFlagField(0, FieldMode.Read, name: "TEIF");
+            transferCompleteFlag = interruptStatusRegister.DefineFlagField(1, FieldMode.Read, name: "TCIF");
+            interruptStatusRegister.DefineFlagField(2, FieldMode.Read, name: "TWIF");
+            interruptStatusRegister.DefineFlagField(3, FieldMode.Read, name: "CAEIF");
+            interruptStatusRegister.DefineFlagField(4, FieldMode.Read, name: "CTCIF");
+            interruptStatusRegister.DefineFlagField(5, FieldMode.Read, name: "CEIF");
+
+
+            var interruptFlagClearRegister = new DoubleWordRegister(this).WithFlag(1, FieldMode.Read | FieldMode.WriteOneToClear, name: "CTCIF", writeCallback: (_, val) => { 
+                if(val) IRQ.Unset(); transferCompleteFlag.Value = false; 
```

Comment:
I believe you wanted braces here, like in 4debe7d58e41f3ab351e4c30044f799c3d704c14

Is that correct?